src/Development/IDE/Core/Rules.hs

```diff
@@ -526,11 +538,22 @@ getModIfaceRule = define $ \GetModIface f -> do
             tmr <- use TypeCheck f
             return ([], extract tmr)
           | otherwise -> do
-            -- Otherwise the interface file does not exist or is out of date. Invoke typechecking directly to update it without incurring a dependency on the typecheck rule.
-            (diags, tmr) <- typeCheckRuleDefinition f DoGenerateInterfaceFiles
-            -- Bang pattern is important to avoid leaking 'tmr'
-            let !res = extract tmr
-            return (diags, res)
+            -- the interface file does not exist or is out of date.
+            -- Invoke typechecking directly to update it without incurring a dependency
+            -- on the parsed module and the typecheck rules
+            hsc <- hscEnv <$> use_ GhcSession f
+            opt <- getIdeOptions
+            (_, contents) <- getFileContents f
+            -- Embed --haddocks in the interface file
+            hsc <- pure hsc{hsc_dflags = gopt_set (hsc_dflags hsc) Opt_Haddock}
```

Comment:
Why do we set this here and not where we set all the other options? Also seems unrelated to the patch as a whole.

Reply:
We want to embed haddocks in the interface files, so we set `Opt_Haddock` on the set of flags that we use for generating them. It's pretty crucial for the patch, without this we cannot skip parsing modules for `getSpanInfos` and `produceCompletions`.

`Opt_Haddock` is not desired globally, see https://github.com/digital-asset/ghcide/pull/350#discussion_r370878197